src/lightning_lite/plugins/environments/slurm_environment.py

```diff
@@ -141,3 +144,19 @@ def resolve_root_node_address(nodes: str) -> str:
         nodes = re.sub(r"\[(.*?)[,-].*\]", "\\1", nodes)  # Take the first node of every node range
         nodes = re.sub(r"\[(.*?)\]", "\\1", nodes)  # handle special case where node range is single number
         return nodes.split(" ")[0].split(",")[0]
+
+    @staticmethod
+    def _validate_srun_variables():
+        """Checks for conflicting or incorrectly set variables set through `srun` and raises a useful error
+        message.
+
+        Right now, we only check for the most commond user errors. See `the srun docs
+        <https://slurm.schedmd.com/srun.html>`_ for a complete list of supported srun variables.
+        """
+
+        ntasks = int(os.environ.get("SLURM_NTASKS", "1"))
+        if ntasks > 1:
+            raise rank_zero_warn(
+                f"You set `--ntasks={ntasks}` in your SLURM bash script, but this variable is not supported. Please use"
+                f" `--ntasks-per-node={ntasks}` instead."
+            )
```

Comment:
@Queuecumber Sorry for pinging you, but as a slurm user you might know this. 
1. Should this be an error? For now I made it a warning as I am not 100% sure if I break someone's code.
2. Would it be useful to also provide a warning/error when SLURM_CPUS_PER_TASK is set, with an error similar as in #10150?

Reply:
I won't claim any special knowledge of slurm outside whatever I need to run my experiments, but my understanding is that `ntasks` and `ntasks-per-node` can both be set and then `ntasks-per-node` functions as a maximum. Why you would want to do that and frankly what that even means I don't know but technically it's "allowed". So in that case I suppose a loud warning is best so that the one person who uses both those options and lightning doesn't have their code crash. 

That said, it may not be the best idea to commit to slurm validation. 

I always viewed the slurm multiprocessing detection/checkpointing/requeueing support in lightning as nice-haves with a limited scope and I think that fits nicely with the rest of what lightning does in terms of making your life easier without being too invasive. For example, lightning isn't writing the sbatch file and submitting the job for you; that would be out of scope.

It may be worth discussing if this kind of validation, too, is out of scope. I won't say that I think it definitely is but it might be worth discussing it internally. Validating all the permutations of slurm options and which ones conflict and how they change based on context is extremely complex and there are other options besides these that people may want validated. The other way to do it is to just go all-in and implement something like submitit which creates the sbatch file based on the trainer config, and that way you have complete control of what options are being used so you wouldn't need any validation logic.

Just something to think about

Reply:
Greatly appreciate the quick response!

>  but my understanding is that ntasks and ntasks-per-node can both be set and then ntasks-per-node functions as a maximum

I would say yes, that's also my understanding that this is allowed in general. But I don't see that being useful or compatible with Lightning Trainer at the moment. Right now, there is no elasticity and we require each node to run the same number of processes. I don't see that changing anytime soon. Since that number set there must also be exactly the same as in the Trainer(devices=x), I don't see how setting both `ntasks-per-node` and `ntasks` to different values is valid. An example of this mistake can be seen here: #14986, and similar questions have been asked in the past and on slack. 

> Validating all the permutations of slurm options and which ones conflict and how they change based on context is extremely complex and there are other options besides these that people may want validated. 

Yeah, I don't think that's feasible. That's why I wrote in the comment in the code that we only intend to catch the most frequent user mistakes. 

> The other way to do it is to just go all-in and implement something like submitit which creates the sbatch file based on the trainer config, and that way you have complete control of what options are being used so you wouldn't need any validation logic.

Out of scope of Lightning I think. I would rather invest in a smooth interaction with submitit and Lightning.

Reply:
In that case I think crashing the job does make sense